app/services/activity_logger.py

```diff
@@ -40,26 +40,28 @@
     wait_jitter=3.0
 )
 async def publish_event(event: SystemEventBaseModel, topic_name: str):
-    client = pubsub_v1.PublisherClient()
-    # Get the topic
-    topic = client.topic_path(settings.GCP_PROJECT_ID, topic_name)
-    # Prepare the payload
-    binary_payload = json.dumps(event.dict(), default=str).encode("utf-8")
-
-    logger.debug(f"Sending event {event} to PubSub topic {topic_name}..")
-    try:  # Send to pubsub
-        # Make the request
-        response = client.publish(topic=topic, data=binary_payload)
-        logger.debug(f"Published message: {binary_payload}")
-    except Exception as e:
-        logger.exception(
-            f"Error publishing system event to topic {topic_name}: {e}. This will be retried."
-        )
-        raise e
-    else:
-        logger.debug(f"System event {event} published successfully.")
-        logger.debug(f"GCP PubSub response: {response.result()}")
-        return response.result()
+    timeout_settings = aiohttp.ClientTimeout(total=10.0)
```

Comment:
We should probably increase this timeout to 30-60 secs. Some PubSub responses can get slower when making many concurrent requests. I should fix that in the template but you can patch it here for testing if you want.

Reply:
and I think we could optimize that function too, to avoid creating a new session every time an event is published